wfdb/io/_signal.py

```diff
@@ -1875,7 +1875,22 @@ def _rd_compressed_file(
             start_samp = start_frame * samps_per_frame[0]
             end_samp = end_frame * samps_per_frame[0]
             sf.seek(start_samp + sample_offset)
-            sig_data = sf.read(end_samp - start_samp, dtype=read_dtype)
+
+            # We could do this:
+            #  sig_data = sf.read(end_samp - start_samp, dtype=read_dtype)
+            # However, sf.read fails for huge blocks (over 2**24 total
+            # samples) due to a bug in libsndfile.  So read the data
+            # in chunks instead.
+            n_samp = end_samp - start_samp
+            sig_data = np.empty((n_samp, n_sig), dtype=read_dtype)
+            CHUNK_SIZE = 1024 * 1024
+            for chunk_start in range(0, n_samp, CHUNK_SIZE):
+                chunk_end = chunk_start + CHUNK_SIZE
+                chunk_data = sf.read(out=sig_data[chunk_start:chunk_end])
+                samples_read = chunk_data.shape[0]
+                if samples_read != CHUNK_SIZE:
```

Comment:
What exactly is this block for? Is it just for the last chunk, or potentially earlier?

Why would the shape of `sig_data` need to be changed? Is its first dimension not guaranteed to be `n_samp`?